lib/Internal/Posix/Runner.php

```diff
@@ -156,6 +156,10 @@ public function kill(ProcessHandle $handle)
         if (!\proc_terminate($handle->proc, 9)) { // Forcefully kill the process using SIGKILL.
             throw new ProcessException("Terminating process failed");
         }
+        $handle->pidDeferred->promise()->onResolve(function ($error, $pid) {
+            // ignore errors because process not always detached
+            \posix_kill($pid, 9);
```

Comment:
The function should not call `posix_kill()` if `$pid` is null (i.e., there was an error starting the process).

Reply:
Fixed